### ISSUE TITLE
[Bug]: Build failure on new project

### ISSUE THREAD

**Issue by fbinz** [apivideo/api.video-reactnative-live-stream#65]
### Version

v1.2.3

### Which operating systems have you used?

- [X] Android
- [ ] iOS

### Environment that reproduces the issue

Does not depend on environment

### Is it reproducible in the example application?

No

### RTMP Server

None required

### Reproduction steps

To reproduce:
1. Create new react-native app: `npx react-native init react_native_test`
2. Install package as per documentation: `yarn add @api-video/react-native-livestream`
3. Try do build and run application on android: `yarn android`

### Expected result

A successfully running example application.

### Actual result

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tasks'.
> Could not create task ':api.video_react-native-livestream:compileDebugAndroidTestKotlin'.
   > Cannot use @TaskAction annotation on method AbstractKotlinCompile.execute() because interface org.gradle.api.tasks.incremental.IncrementalTaskInputs is not a valid parameter to an action method.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 5s
error Failed to install the app. Command failed with exit code 1: ./gradlew tasks FAILURE: Build failed with an exception. * What went wrong:
Execution failed for task ':tasks'.
> Could not create task ':api.video_react-native-livestream:compileDebugAndroidTestKotlin'. > Cannot use @TaskAction annotation on method AbstractKotlinCompile.execute() because interface org.gradle.api.tasks.incremental.IncrementalTaskInputs is not a valid parameter to an action method. * Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights. * Get more help at https://help.gradle.org BUILD FAILED in 5s
> Task :gradle-plugin:pluginDescriptors
> Task :gradle-plugin:processResources
> Task :gradle-plugin:compileKotlin
> Task :gradle-plugin:compileJava NO-SOURCE
> Task :gradle-plugin:classes
> Task :gradle-plugin:jar
> Task :gradle-plugin:inspectClassesForKotlinIC
> Task :tasks FAILED
6 actionable tasks: 6 executed.
error Command failed with exit code 1.
```

**Comment by Nonag** [comment]
I tried to get the plugin running with a fresh RN 0.72.x install the other day and can confirm that the app won't build.

**Comment by ThibaultBee** [comment]
Hi,

The project and its dependencies are getting too old.
We will have to update the project but it requires to support both new and old RN architecture. It is not an easy upgrade.
Also, Swift is still not supported with the new architecture (unless on 0.72.2 and xcode 14.3) and native part is unfortunately in Swift. We don't plan to migrate to objective-c because it looks like a backward development.

**Comment by soemarko** [comment]
Any workaround for this? Can I just downgrade RN for now? Which version should I set it to?

**Comment by Veryinheart** [comment]
any update on this? I get  this error on RN version 0.72 and api.video version 1.2.4.

**Comment by ThibaultBee** [comment]
Hi,

We are currently in the process of upgrading to 0.73 (and support new architecture). See https://github.com/apivideo/api.video-reactnative-live-stream/pull/67 instead.

**Comment by Veryinheart** [comment]
> Hi,
> 
> We are currently in the process of upgrading to 0.73 (and support new architecture). See #67 instead.

Hi,
Do you have ETA of this? how long does it take to be fixed? many thanks

**Comment by ThibaultBee** [comment]
The dev is done. I will tackle a last issue with the Android preview.

But... the release is not ready yet. We are expecting feedbacks/tests from the community before releasing. See https://github.com/apivideo/api.video-reactnative-live-stream/pull/67

**Comment by ThibaultBee** [comment]
Could you test with 2.0.0?

**Comment by soemarko** [comment]
Missing type definition?
```
Could not find a declaration file for module '@api.video/react-native-livestream'. '/Users/s/App/Livestream/node_modules/@api.video/react-native-livestream/lib/commonjs/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/api.video__react-native-livestream` if it exists or add a new declaration (.d.ts) file containing `declare module '@api.video/react-native-livestream';
```

And it used to have `LiveStreamMethods` for ref type, is it gone now?

**Comment by ThibaultBee** [comment]
Hi,
Type have been renamed `ApiVideo....`. `LiveStreamMethods` has been renamed `ApiVideoLiveStreamMethods`.
It is in the `[CHANGELOG.md](https://github.com/apivideo/api.video-reactnative-live-stream/blob/main/CHANGELOG.md#200---2024-05-02)`

**Comment by soemarko** [comment]
Gotcha, but the type definition is still missing?

**Comment by ThibaultBee** [comment]
Yes. But fixed in https://github.com/apivideo/api.video-reactnative-live-stream/pull/77

**Comment by ThibaultBee** [comment]
I am closing this issue as the initial issue is resolved with 2.0.0.
